src/OAuthClient/BibClient.php

```diff
@@ -27,14 +30,28 @@ public static function getBibByIdAndSource($bibId = '', $nyplSource)
             $response['data']
         );
 
-        if ($response['statusCode'] !== 200) {
+        // Check statusCode range
+        if ($response['statusCode'] === 200) {
+            return new Bib($response['data']);
+        } elseif ($response['statusCode'] >= 500 && $response['statusCode'] <= 599) {
+            throw new APIException(
```

Comment:
You might want to create a `RetriableAPIException` class that extends `APIException`.